receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,219 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8slogreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8slogreceiver"
+
+import (
+	"fmt"
+
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+)
+
+const (
+	ModeDaemonSetStdout = "daemonset-stdout"
+)
+
+const (
+	DefaultMode        = ModeDaemonSetStdout
+	DefaultHostRoot    = "/host_root"
+	DefaultNodeFromEnv = "K8S_NODE_NAME"
+)
+
+// Config is the configuration of a k8slog receiver
+type Config struct {
+	Discovery SourceConfig  `mapstructure:"discovery"`
+	Extract   ExtractConfig `mapstructure:"extract"`
+
+	// TODO: refactor fileconsumer and add it's config of k8s implementation here.
+}
+
+// ExtractConfig allows specifying how to extract resource attributes from pod.
+type ExtractConfig struct {
+	// Metadata represents the list of metadata fields to extract from pod.
+	// TODO: supported metadata fields and default values.
+	Metadata []string `mapstructure:"metadata"`
+
+	// Annotations represents the rules to extract from pod annotations.
+	Annotations []FieldExtractConfig `mapstructure:"annotations"`
+
+	// Labels represents the rules to extract from pod labels.
+	Labels []FieldExtractConfig `mapstructure:"labels"`
+
+	// Env represents the rules to extract from container environment variables.
+	Env []FieldExtractConfig `mapstructure:"env"`
+
+	// OtelEnv represents the rules to extract from environment variables of otel itself.
+	OtelEnv []FieldExtractConfig `mapstructure:"otel_env"`
```

Comment:
I don't think we should allow extracting from env variables, that feels like a security risk.

Reply:
`OtelEnv` has been removed.
However, `Env` is needed in some cases. For example, if there are some environment variables in the docker image, there is no other way to get it.
Is this really risky? `resourcedetection` supports it, and this component gets k8s node name from environment variables, too.

Reply:
Using an env var to filter feels ok. But this section is for taking fields from the pod and adding them as resource attributes to the log right?

Reply:
Getting attributes from environment variables is useful, and many components support it such as `resoucedetection`

Are there any reports of security risks associated with reading environment variables as strings and adding them to data?

Reply:
The risk is exposing apikey/password information, but that risk is lowered if the config only allows static strings.  If we do allow this I think we can do all env vars with 1 config option, no need to split out OTel env vars.

Reply:
By default, the `env` config is empty, and no variable will be exposed. When a user set this, he should know what he is doing.

It seems that `otel_env` is necessary, so I've removed it in the latest commit.

Reply:
Any updates for this?

Reply:
Lets move forward with this allowed as we've discussed.

Reply:
I also find it useful to have the option to collect ENV_VAR information from the Pods.
That would give us important information about OTEL specific info like `OTEL_SERVICE_NAME`.
Having [`service.name`](https://opentelemetry.io/docs/specs/semconv/resource/#service) populated in log records collected by the k8slog receiver is quite important since it will allow us to correlate logs with traces using the `service.name` value.